src/attngan/code/datasets.py

```diff
@@ -0,0 +1,312 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+
+from nltk.tokenize import RegexpTokenizer
+from collections import defaultdict
+from miscc.config import cfg
+
+import torch
+import torch.utils.data as data
+from torch.autograd import Variable
+import torchvision.transforms as transforms
+
+import os
+import sys
+import numpy as np
+import pandas as pd
+from PIL import Image
+import numpy.random as random
+if sys.version_info[0] == 2:
+    import cPickle as pickle
+else:
+    import pickle
+
+
+def prepare_data(data):
+    imgs, captions, captions_lens, class_ids, keys = data
```

Comment:
Add docstring about what's in `data`

Reply:
- doctstring added
- `dot` files removed